src/tests/shared.rs

```diff
@@ -1231,3 +1231,67 @@ pub(crate) async fn wallet_state_has_all_valid_mps(
 
     true
 }
+
+// recursively copy source dir to destination
+pub fn copy_dir_recursive(source: &PathBuf, destination: &PathBuf) -> std::io::Result<()> {
+    if !source.is_dir() {
+        return Err(std::io::Error::other("Source is not a directory"));
+    }
+    std::fs::create_dir_all(destination)?;
+    for entry in std::fs::read_dir(source)? {
+        let entry = entry?;
+        let dest_path = &destination.join(entry.file_name());
+        if entry.path().is_dir() {
+            copy_dir_recursive(&entry.path(), dest_path)?;
+        } else {
+            std::fs::copy(entry.path(), dest_path)?;
+        }
+    }
+    Ok(())
+}
```

Comment:
Rebase artifact? This is already in `lib.rs`.

Reply:
I think so, yes.  I recall a conflict about it.  will remove from here.

Reply:
removed dup fn in 1b9a52926cb7755fc853048ec76c5d066a32d47d